### DISCUSSION TITLE
Lentidão Utilizando Next.js

### DISCUSSION THREAD

**Discussion by arthurtg11** [frontendbr/forum#2404] [Perguntas e Respostas]
Boa noite, estou com um problema na minha aplicação com o next.js, se eu faço o build da aplicação e subo com next start, ao acessar o localhost, fica super rapido, porém  ao iniciar com yarn dev, a pagina demora uns 3s para carregar, quando utilizo o Router.push(), é instantâneo, porém ao mudar a url alterando a pagina, exemplo, do /login para /main, demora em torno de 3s para carregar.

Console do VSCODE ao startar a aplicação:
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled client and server successfully in 2.3s (1301 modules)
wait  - compiling / (client and server)...
wait  - compiling /_error (client and server)...
event - compiled client and server successfully in 531 ms (1305 modules)
(node:24672) ExperimentalWarning: AbortController is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
warn  - Fast Refresh had to perform a full reload. Read more: https://nextjs.org/docs/messages/fast-refresh-reload

**Comment by ninetails** [comment]
Uma das diferenças entre utilizar `next dev` e utilizar o `build` daí `next start` é que em modo de desenvolvimento a tua aplicação compila toda tua aplicação em caso de mudanças de arquivos localmente quando bate numa rota nova.

A rota que foi aberta quando vc subiu a aplicação e mandou de exemplo de saída do console. Veja que nela 1301 arquivos são compilados na rota aberta. Se você tá com TypeScript, serão 1301 arquivos (x2 pq o next vai compilar a rota no servidor e o que vai pro usuário no client) que será passado o compilador de typescript. Ou do babel, caso não utilize TS.

O que eu diria é: assegure que esteja sendo usado o SWC (novo compilador em Rust), se ainda persistir diria pra tentar modularizar sua aplicação para uma arquitetura que vc possa trabalhar com pedaços já pré-compilados (transformar partes da tua aplicação em pacotes separados e já transpilados) pra evitar que tenha que compilar partes do código que não precise no momento.
